.github/scripts/free-disk-space.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+echo "=============================================================================="
+echo "Freeing up disk space on CI system"
+echo "=============================================================================="
+
+echo "Listing 100 largest packages"
+dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -n | tail -n 100
+df -h
+echo "Removing large packages"
+sudo apt-get remove -y '^dotnet-.*'
+sudo apt-get remove -y '^llvm-.*'
+sudo apt-get remove -y 'php.*'
+sudo apt-get remove -y azure-cli google-cloud-* google-chrome-stable firefox powershell mono* microsoft-edge-stable temurin-*
```

Comment:
The fact that this is needed makes me worry that what we're doing isn't really supported and might not be very future-proof. How big is the image, and how many extra copies do we end up creating compared to the baseline?

Reply:
It looks like the image is ~1GiB, and according to the [Github docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories), ubuntu-latest has 150 GiB.

Reply:
And i don't think it's 150GiB, docs might be wrong there. But even still, there's 20GB free on each runner given whatever boiler plate they install on the runner to start with, which should be plenty.

By my count I think there would be 3 copies of the image: the locally built image, the copy that gets pushed to a registry, which happens to be local, and the compressed tarball made from the registry image. And idk if any other temp copies are made internally. Still doesn't really add up, not sure why I hit a disk space limit at one point.

Reply:
Not sure how I missed this before, but there's a much better way to produce the tarball
https://docs.docker.com/build/ci/github-actions/share-image-jobs/#:~:text=outputs%3A%20type%3Ddocker%2Cdest%3D/tmp/myimage.tar

Testing it now

Reply:
Removed this script